app/models/principal.rb

```diff
@@ -71,8 +71,32 @@ def full_name
     "#{first_name} #{last_name}"
   end
 
+  def next_onboarding_action
+    registered_firms = main_firm_with_trading_names.registered
+
+    if registered_firms.empty?
+      :complete_a_firm
+    elsif needs_advisers?(registered_firms)
+      :complete_an_adviser
+    else
+      :onboarded
+    end
+  end
+
   private
 
+  def needs_advisers?(firm_list)
+     no_remote_firms?(firm_list) && no_advisers_for?(firm_list)
```

Comment:
I agree with Pete. We need advisers if there **are any** firms that do offer face-to-face advice (referred to as `in_person_advice` in `firm.rb`). I really think we should positively check for the type that we know does need advisers.

```
any_in_person_advice_firms? && no_advisers_for?(firm_list)
```